Serenity/JS 3.0 - LastResponse.body<Interface>.location / .description not accessible

I create this issue to separate it from #1342.

Imagine an interface:
```ts
    interface Product {
        id: number;
        name: string;
        description?: string;
        location?: string;
    }
```

And a test:
```ts
 it('enables access to the response body if interface contains a property called description or location', () =>
       actor.attemptsTo(
       Send.a(GetRequest.to(productDetails)),
       Ensure.that(LastResponse.body<Product>().name, equals('apple')),
       Ensure.that(LastResponse.body<Product>().description, equals('a delicious fruit')),
       Ensure.that(LastResponse.body<Product>().location, equals('fruit bowl')),
       )
  );
```

Then `.description` and `.location` is not accessible as expected. It access the constructor of the `Activity` class instead of the response object.

```error TS2445: Property 'description' is protected and only accessible within class 'Activity' and its subclasses.```

Created a branch in my fork that reproduces this issue: https://github.com/viper3400/serenity-js/tree/repro/1342

https://github.com/viper3400/serenity-js/blob/db4340201fefa130743884fed43d284a2aeb06c6/packages/rest/spec/screenplay/questions/LastResponse.spec.ts#L60

--- a/packages/core/spec/screenplay/QuestionAdapter.spec.ts
+++ b/packages/core/spec/screenplay/QuestionAdapter.spec.ts
@@ -425,6 +425,8 @@ describe('Question', () => {
                         name: string;
                         caller: string;
                         arguments: boolean;
+                        description: string;
+                        location: string;
                     }
 
                     const Response = () =>
@@ -433,8 +435,8 @@ describe('Question', () => {
                                 // "caller" and "argument" are here to show how a proxy resolves conflicts
                                 // between a built-in fields like function.caller and function.arguments
                                 // and properties of the wrapped object of the same name
-                                { name: 'Alice',    caller: 'Bob',      arguments: false },
-                                { name: 'Bob',      caller: 'Alice',    arguments: true  },
+                                { name: 'Alice',    caller: 'Bob',      arguments: false,   location: 'London',     description: 'first actor' },
+                                { name: 'Bob',      caller: 'Alice',    arguments: true,    location: 'New York',   description: 'second actor'  },
                             ]
                         }));
 
@@ -468,6 +470,22 @@ describe('Question', () => {
                         const result: string = await actor.answer(name);
 
                         expect(result).to.equal('Alice');
+                    });
+
+                    it('proxies "description"', async () => {
+                        const description: Question<Promise<string>> = Response().users[0].description;
+
+                        const result: string = await actor.answer(description);
+
+                        expect(result).to.equal('first actor');
+                    });
+
+                    it('proxies "location"', async () => {
+                        const location: Question<Promise<string>> = Response().users[0].location;
+
+                        const result: string = await actor.answer(location);
+
+                        expect(result).to.equal('London');
                     });
                 });
             });

--- a/packages/core/src/screenplay/Activity.ts
+++ b/packages/core/src/screenplay/Activity.ts
@@ -18,18 +18,22 @@ import { AnswersQuestions, PerformsActivities, UsesAbilities } from './actor';
 export abstract class Activity {
 
     private static errorStackParser = new ErrorStackParser();
+    readonly #description: string;
+    readonly #location: FileSystemLocation;
 
     constructor(
-        protected readonly description: string,
-        private readonly location: FileSystemLocation = Activity.callerLocation(5)
+        description: string,
+        location: FileSystemLocation = Activity.callerLocation(5)
     ) {
+        this.#description = description;
+        this.#location = location;
     }
 
     /**
      * Returns the location where this {@apilink Activity} was instantiated.
      */
     instantiationLocation(): FileSystemLocation {
-        return this.location;
+        return this.#location;
     }
 
     /**
@@ -54,7 +58,7 @@ export abstract class Activity {
      * For example, `#actor clicks on a button` becomes `Wendy clicks on a button`.
      */
     toString(): string {
-        return this.description;
+        return this.#description;
     }
 
     protected static callerLocation(frameOffset: number): FileSystemLocation {

--- a/packages/core/src/screenplay/Question.ts
+++ b/packages/core/src/screenplay/Question.ts
@@ -238,6 +238,7 @@ export abstract class Question<T> {
                         && key === 'describedAs';   // `describedAs` returns `this`, which must be bound to proxy itself to allow proxy chaining
 
                     if (mustAllowProxyChaining) {
+                        // see https://javascript.info/proxy#proxy-limitations
                         return field.bind(receiver)
                     }
 

--- a/packages/core/src/screenplay/Question.ts
+++ b/packages/core/src/screenplay/Question.ts
@@ -229,7 +229,21 @@ export abstract class Question<T> {
                 }
 
                 if (key in target) {
-                    return Reflect.get(target, key);
+
+                    const field = Reflect.get(target, key);
+
+                    const isFunction = typeof field == 'function'
+                    const mustAllowProxyChaining = isFunction
+                        && target instanceof QuestionStatement
+                        && key === 'describedAs';   // `describedAs` returns `this`, which must be bound to proxy itself to allow proxy chaining
+
+                    if (mustAllowProxyChaining) {
+                        return field.bind(receiver)
+                    }
+
+                    return isFunction
+                        ? field.bind(target)
+                        : field;
                 }
 
                 if (key === 'then') {

--- a/packages/web/src/screenplay/interactions/ExecuteScript.ts
+++ b/packages/web/src/screenplay/interactions/ExecuteScript.ts
@@ -248,7 +248,7 @@ class ExecuteAsynchronousScript extends ExecuteScriptWithArguments {
         return new ExecuteAsynchronousScript(
             args.length > 0
                 ? d `#actor executes an asynchronous script with arguments: ${ args }`
-                : this.description,
+                : this.toString(),
             this.script,
             args,
         );
@@ -320,7 +320,7 @@ class ExecuteSynchronousScript extends ExecuteScriptWithArguments {
         return new ExecuteSynchronousScript(
             args.length > 0
                 ? d `#actor executes a synchronous script with arguments: ${ args }`
-                : this.description,
+                : this.toString(),
             this.script,
             args,
         );

